libzerocash/ZerocashParams.h

```diff
@@ -24,29 +24,34 @@ class ZerocashParams {
 public:
     typedef default_r1cs_ppzksnark_pp zerocash_pp;
 
-	ZerocashParams(const unsigned int tree_depth);
-	ZerocashParams(zerocash_pour_proving_key<zerocash_pp>* p_pk_1,
-                   zerocash_pour_verification_key<zerocash_pp>* p_vk_1);
-
-    ZerocashParams(const unsigned int tree_depth,
-                   std::string pathToProvingParams,
-                   std::string pathToVerificationParams);
+    ZerocashParams(
+        const unsigned int tree_depth,
+        zerocash_pour_keypair<ZerocashParams::zerocash_pp> *keypair
+    );
+
+    ZerocashParams(
+        const unsigned int tree_depth,
+        zerocash_pour_proving_key<ZerocashParams::zerocash_pp>* p_pk_1,
+        zerocash_pour_verification_key<ZerocashParams::zerocash_pp>* p_vk_1
+    );
+
+    const zerocash_pour_proving_key<zerocash_pp>& getProvingKey();
+    const zerocash_pour_verification_key<zerocash_pp>& getVerificationKey();
+    ~ZerocashParams();
 
-    const zerocash_pour_proving_key<zerocash_pp>& getProvingKey(const int version);
+    static const size_t numPourInputs = 2;
+    static const size_t numPourOutputs = 2;
```

Comment:
[does not block] These are derived from the parameters, so just like ``tree_depth`` it would be nice if they were computed from the parameters, or the parameter loading code would fail if these values weren't set appropriately in the serializaiton.